src/CommitmentsV2/SFA.DAS.CommitmentsV2/Models/ApprenticeshipBase.cs

```diff
@@ -25,6 +25,7 @@ protected ApprenticeshipBase()
         public bool TrainingCourseVersionConfirmed { get; set; }
         public string TrainingCourseOption { get; set; }
         public string StandardUId { get; set; }
+        public DeliveryModel? DeliveryModel { get; set; }
```

Comment:
I think this should be non-nullable in the entity.  

Delivery model is mandatory and it's much cleaner for the domain code to only see Normal or Flexible (or other future values).  From a domain perspective there are no nulls, so domain code shouldn't have to worry about it

We should put the logic around "null means normal" into the Entity Framework code, or hide it away inside the entity.  If we let the entity contain a null then it requires all the domain code need to know that null also means Normal, and require checks if model == null || model == Normal .

Reply:
Interestingly, I made it a non nullable entity in the PA-99 branch, but fell into a trap of when EF was generating a SQL query and returning the results, EF assumed the value would always have a value present and the query failed. So I ended up making it Nullable again.